[OBS] nil pointer dereference on bucket creation

<!--
Hi there, thank you for opening an issue.
-->

# Terraform Version
```
Terraform v0.13.5
+ provider registry.terraform.io/opentelekomcloud/opentelekomcloud v1.21.1
```

# Affected Resource(s)
`opentelekomcloud_s3_bucket`

# Terraform Configuration Files
```hcl
terraform {
  required_providers {
    opentelekomcloud = {
      source  = "opentelekomcloud/opentelekomcloud"
      version = "~> 1.21.5"
    }
  }
  required_version = ">= 0.13"
}

provider opentelekomcloud {
  auth_url = "https://iam.eu-de.otc.t-systems.com/v3/"
  domain_name = "...448"
  access_key = "..."
  secret_key = "..."
}

resource "opentelekomcloud_s3_bucket" "bk" {
  bucket = "my-tf-test-bucket-akachuri"
  acl = "private"
}

```

# Debug Output/Panic Output
```
panic: runtime error: invalid memory address or nil pointer dereference
2020-11-05T13:46:50.833+0300 [DEBUG] plugin.terraform-provider-opentelekomcloud_v1.21.1: [signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x108bede]
2020-11-05T13:46:50.833+0300 [DEBUG] plugin.terraform-provider-opentelekomcloud_v1.21.1: 
2020-11-05T13:46:50.833+0300 [DEBUG] plugin.terraform-provider-opentelekomcloud_v1.21.1: goroutine 56 [running]:
2020-11-05T13:46:50.833+0300 [DEBUG] plugin.terraform-provider-opentelekomcloud_v1.21.1: github.com/opentelekomcloud/terraform-provider-opentelekomcloud/opentelekomcloud.(*Config).computeS3conn(0xc00008e4e0, 0x0, 0x0, 0x0, 0xb, 0x1
...
````

# Steps to Reproduce
1. `terraform apply`

# Expected Behavior
Bucket created

# Actual Behavior
`!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!`

# Important Factoids
Not reproduced after addding `tenant_name = "eu-de"` to provider configuration

--- a/opentelekomcloud/config.go
+++ b/opentelekomcloud/config.go
@@ -530,14 +530,15 @@ func (c *Config) computeS3conn(region string) (*s3.S3, error) {
 		return nil, fmt.Errorf("Missing credentials for Swift S3 Provider, need access_key and secret_key values for provider.")
 	}
 
-	client, err := openstack.NewImageServiceV2(c.HwClient, golangsdk.EndpointOpts{
+	client, err := openstack.NewOBSService(c.HwClient, golangsdk.EndpointOpts{
 		Region:       region,
 		Availability: c.getHwEndpointType(),
 	})
-	// Bit of a hack, seems the only way to compute this.
-	endpoint := strings.Replace(client.Endpoint, "//ims", "//obs", 1)
+	if err != nil {
+		return nil, err
+	}
 
-	awsS3Sess := c.s3sess.Copy(&aws.Config{Endpoint: aws.String(endpoint)})
+	awsS3Sess := c.s3sess.Copy(&aws.Config{Endpoint: aws.String(client.Endpoint)})
 	s3conn := s3.New(awsS3Sess)
 
 	return s3conn, err

--- a/opentelekomcloud/resource_opentelekomcloud_s3_bucket.go
+++ b/opentelekomcloud/resource_opentelekomcloud_s3_bucket.go
@@ -283,7 +283,8 @@ func resourceS3Bucket() *schema.Resource {
 
 func resourceS3BucketCreate(d *schema.ResourceData, meta interface{}) error {
 	config := meta.(*Config)
-	s3conn, err := config.computeS3conn(GetRegion(d, config))
+	region := GetRegion(d, config)
+	s3conn, err := config.computeS3conn(region)
 	if err != nil {
 		return fmt.Errorf("Error creating OpenTelekomCloud s3 client: %s", err)
 	}
@@ -307,15 +308,9 @@ func resourceS3BucketCreate(d *schema.ResourceData, meta interface{}) error {
 		ACL:    aws.String(acl),
 	}
 
-	var awsRegion string
-	if region, ok := d.GetOk("region"); ok {
-		awsRegion = region.(string)
-	} else {
-		awsRegion = meta.(*Config).Region
-	}
-	log.Printf("[DEBUG] S3 bucket create: %s, using region: %s", bucket, awsRegion)
+	log.Printf("[DEBUG] S3 bucket create: %s, using region: %s", bucket, region)
 
-	if err := validateS3BucketName(bucket, awsRegion); err != nil {
+	if err := validateS3BucketName(bucket, region); err != nil {
 		return fmt.Errorf("Error validating S3 bucket name: %s", err)
 	}
 
@@ -690,7 +685,7 @@ func resourceS3BucketRead(d *schema.ResourceData, meta interface{}) error {
 				}
 				rule["noncurrent_version_expiration"] = schema.NewSet(expirationHash, []interface{}{e})
 			}
-			//// transition
+			// transition
 			if len(lifecycleRule.Transitions) > 0 {
 				transitions := make([]interface{}, 0, len(lifecycleRule.Transitions))
 				for _, v := range lifecycleRule.Transitions {
@@ -787,7 +782,7 @@ func resourceS3BucketRead(d *schema.ResourceData, meta interface{}) error {
 	}
 
 	// UNUSED?
-	//d.Set("arn", fmt.Sprintf("arn:%s:s3:::%s", meta.(*Config).partition, d.Id()))
+	// d.Set("arn", fmt.Sprintf("arn:%s:s3:::%s", meta.(*Config).partition, d.Id()))
 
 	return nil
 }
@@ -1224,7 +1219,7 @@ func resourceAwsS3BucketLifecycleUpdate(s3conn *s3.S3, d *schema.ResourceData) e
 
 	lifecycleRules := d.Get("lifecycle_rule").([]interface{})
 
-	//fmt.Printf("lifecycleRules=%+v.\n", lifecycleRules)
+	// fmt.Printf("lifecycleRules=%+v.\n", lifecycleRules)
 	if len(lifecycleRules) == 0 {
 		i := &s3.DeleteBucketLifecycleInput{
 			Bucket: aws.String(bucket),
@@ -1308,14 +1303,14 @@ func resourceAwsS3BucketLifecycleUpdate(s3conn *s3.S3, d *schema.ResourceData) e
 		rules = append(rules, rule)
 	}
 
-	//fmt.Printf("Rules=%+v.\n", rules)
+	// fmt.Printf("Rules=%+v.\n", rules)
 	i := &s3.PutBucketLifecycleConfigurationInput{
 		Bucket: aws.String(bucket),
 		LifecycleConfiguration: &s3.BucketLifecycleConfiguration{
 			Rules: rules,
 		},
 	}
-	//fmt.Printf("PutBucketLifecycleConfigurationInput=%+v.\n", i)
+	// fmt.Printf("PutBucketLifecycleConfigurationInput=%+v.\n", i)
 
 	err := resource.Retry(1*time.Minute, func() *resource.RetryError {
 		if _, err := s3conn.PutBucketLifecycleConfiguration(i); err != nil {

